src/components/core/datetime/date-adapter.ts

```diff
@@ -120,46 +118,119 @@ export interface DateAdapter<T = any> {
    * @param date The starting date.
    * @param months The number of months to add.
    */
-  addCalendarMonths: (date: T, months: number) => T;
+  public abstract addCalendarMonths(date: T, months: number): T;
 
   /** Creates a new date by adding the number of provided `days` to the provided `date`.
    * @param date The starting date.
    * @param days The number of days to add.
    */
-  addCalendarDays: (date: T, days: number) => T;
-
-  /**
-   * Calculates the first year that will be shown in the year selection panel.
-   * @param activeDate The active date.
-   * @param minDate The minimum date, if set.
-   * @param maxDate The maximum date, if set.
-   */
-  getStartValueYearView: (activeDate: T, minDate?: T | null, maxDate?: T | null) => number;
+  public abstract addCalendarDays(date: T, days: number): T;
 
   /** Get the date in the local format.
    * @param date The date to format
    * @returns The `date` in the local format as string.
    */
-  getAccessibilityFormatDate: (date: T | string) => string;
+  public abstract getAccessibilityFormatDate(date: T | string): string;
 
   /** Get the given string as Date.
    * @param value The date in the format DD.MM.YYYY.
    * @param now The current date as Date.
    */
-  parseDate: (value: string | null | undefined, now: Date) => T | undefined;
+  public abstract parse(value: string | null | undefined, now: Date): T | undefined;
 
   /** Format the given Date as string.
    * @param value The date to format.
    */
-  format: (date: T | null | undefined) => string;
+  public abstract format(date: T | null | undefined): string;
 
   /** Checks whether the given `obj` is a Date.
    * @param obj The object to check.
    */
-  isDateInstance: (obj: any) => boolean;
+  public abstract isDateInstance(obj: any): boolean;
+
+  /**
+   * Gets date instance that is not valid.
```

Comment:
hard to understand the sense for me.hmm

Reply:
This is a method that generates an intentionally invalid date.

Reply:
You should once explain me the sense ;). but surely ok to merge without knowing exactly :)